packages/trading-client/src/handlers/coordinator.ts

```diff
@@ -0,0 +1,245 @@
+            case ExchangeMethods.CancelOrder:
+                return transactionEncoder.cancelOrderTx(order);
+            case BatchExchangeMethods.BatchFillOrders:
+                return transactionEncoder.batchFillOrdersTx(order, assetFillAmount);
+            case BatchExchangeMethods.BatchFillOrKillOrders:
+                return transactionEncoder.batchFillOrKillOrdersTx(order, assetFillAmount);
+            case BatchExchangeMethods.BatchFillOrdersNoThrow:
+                return transactionEncoder.batchFillOrdersNoThrowTx(order, assetFillAmount);
+            case BatchExchangeMethods.MarketSellOrders:
+                return transactionEncoder.marketSellOrdersTx(order, assetFillAmount);
+            case BatchExchangeMethods.MarketSellOrdersNoThrow:
+                return transactionEncoder.marketSellOrdersNoThrowTx(order, assetFillAmount);
+            case BatchExchangeMethods.MarketBuyOrders:
+                return transactionEncoder.marketBuyOrdersTx(order, assetFillAmount);
+            case BatchExchangeMethods.MarketBuyOrdersNoThrow:
+                return transactionEncoder.marketBuyOrdersNoThrowTx(order, assetFillAmount);
+            case BatchExchangeMethods.BatchCancelOrders:
+                return transactionEncoder.batchCancelOrdersTx(order);
+            default:
+                throw new Error('Invalid method name');
+        }
+    }
+    public getCoordinatorInstance(address: string): CoordinatorContract {
+        if (this._coordinatorInstances[address] === undefined) {
+            this._coordinatorInstances[address] = new CoordinatorContract(
+                Coordinator.compilerOutput.abi,
+                address,
+                this._web3Wrapper.getProvider(),
+                this._web3Wrapper.getContractDefaults(),
+            );
+        }
+        return this._coordinatorInstances[address];
+    }
+    public async updateCoordinatorEndpointsAsync(): Promise<void> {
+        if (Object.keys(this._coordinatorEndpoints) !== Object.keys(networkInfoByCoordinatorAddress)) {
+            this._coordinatorEndpoints = await this._getCoordinatorEndpointsAsync();
+        }
+    }
+    // todo: does it return all endpoints when calling await? test by calling constructor and checking for endpoints
+    private async _getCoordinatorEndpointsAsync(): Promise<{ [adddress: string]: string }> {
+        const result: { [adddress: string]: string } = {};
+        Object.keys(networkInfoByCoordinatorAddress).forEach(async address => {
+            const registryAddress = networkInfoByCoordinatorAddress[address].registryAddress;
+            const registryInstance = new CoordinatorRegistryContract(
+                CoordinatorRegistry.compilerOutput.abi,
+                registryAddress,
+                this._web3Wrapper.getProvider(),
+                this._web3Wrapper.getContractDefaults(),
+            );
+            const endpoint = await registryInstance.getCoordinatorEndpoint.callAsync(address, {});
```

Comment:
I think you misunderstood the role of the coordinator registry. It maps a coordinator's `feeRecipientAddress` to an http endpoint. You need to query by `feeRecipientAddress`. Does that make sense?